### DISCUSSION TITLE
How can I sort data in ListBox?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#11460] [Q&A] [answered]
I want to sort data in ListBox. Everytime it adds or removes data, it needs sorting again.

In WPF I can achieve this by ListBox like this:
`ListBox.Items.SortDescriptions.Add(new SortDescription("FirstName",ListSortDirection.Ascending));`

However, there is no SortDescriptions in the Listbox in Avalonia.

How can I solve this? Thank you.

**Comment by maxkatz6** [comment, marked as answer]
You sort original data source that you bind to. It supports INotifyColletionChanged, so any changes in ObservableCollection will be visible in the ListBox as well. Advanced solutions like DynamicData are also supported https://github.com/reactivemarbles/DynamicData.

**Reply by kyuranger** [reply]
> You sort original data source that you bind to. It supports INotifyColletionChanged, so any changes in ObservableCollection will be visible in the ListBox as well. Advanced solutions like DynamicData are also supported https://github.com/reactivemarbles/DynamicData.

Based on your suggestion of directly sorting an ObservableCollection, I found this discussion on StackOverFlow(https://stackoverflow.com/questions/1945461/how-do-i-sort-an-observable-collection). The method mentioned in the confirmed answer above is also worth using. I have spent a lot of time learning how to use DynamicData, but it has few tutorials and it is very troublesome to use. So I think it would be more appropriate and simple for me to directly sort the ObservableCollection.

**Reply by maxkatz6** [reply]
Yes, DynamicData is very powerful, but also quite difficult to use. ObservableCollection is a great solution for simple tasks.
